esp/esdllib/esdl_script.cpp

```diff
@@ -2310,6 +2340,318 @@ class CEsdlTransformOperationFunction : public CEsdlTransformOperationWithChildr
+    enum Type { Text, Signed, Unsigned, Decimal };
+    enum Mode { Append, Set };
+    TxSummaryCoreInfo     m_info;
+    Owned<ICompiledXpath> m_value;
+    Type                  m_type = Text;
+    Mode                  m_mode = Append;
+
+public:
+    CEsdlTransformOperationTxSummaryValue(IXmlPullParser& xpp, StartTag& stag, const StringBuffer& prefix)
+        : CEsdlTransformOperationWithoutChildren(xpp, stag, prefix)
+        , m_info(stag, m_tagname, m_traceName, m_ignoreCodingErrors)
+    {
+        m_value.setown(compileOptionalXpath(stag.getValue("select")));
+        if (!m_value)
+            esdlOperationError(ESDL_SCRIPT_MissingOperationAttr, m_tagname, "without select", m_traceName, !m_ignoreCodingErrors);
+        const char* type = stag.getValue("type");
+        if (!isEmptyString(type))
+        {
+            if (strieq(type, "signed"))
+                m_type = Signed;
+            else if (strieq(type, "unsigned"))
+                m_type = Unsigned;
+            else if (strieq(type, "decimal"))
+                m_type = Decimal;
+            else if (!strieq(type, "text"))
+                esdlOperationError(ESDL_SCRIPT_InvalidOperationAttr, m_tagname, "invalid type", m_traceName, !m_ignoreCodingErrors);
+        }
+        const char* mode = stag.getValue("mode");
+        if (!isEmptyString(mode))
+        {
+            if (strieq(mode, "set"))
+                m_mode = Set;
+            else if (!strieq(mode, "append"))
+                esdlOperationError(ESDL_SCRIPT_InvalidOperationAttr, m_tagname, VStringBuffer("invalid mode '%s'", mode), m_traceName, !m_ignoreCodingErrors);
+        }
+    }
+
+    virtual ~CEsdlTransformOperationTxSummaryValue()
+    {
+    }
+
+    void injectText(const char* value, CTxSummary* txSummary) const
+    {
+        switch (m_mode)
+        {
+        case Append:
+            if (!txSummary->append(m_info.m_name, value, m_info.m_level, m_info.m_groups))
+                UWARNLOG("script operation %s did not append '%s=%s'; name is malformed or is in use", m_tagname.str(), m_info.m_name.str(), value);
+            break;
+        case Set:
+            if (!txSummary->set(m_info.m_name, value, m_info.m_level, m_info.m_groups))
+                UWARNLOG("script operation %s did not set '%s=%s'; name is malformed or identifies a cumulative timer", m_tagname.str(), m_info.m_name.str(), value);
+            break;
+        default:
+            break;
+        }
+    }
+    void injectSigned(const char* rawValue, CTxSummary* txSummary) const
+    {
+        signed long long sll;
+        if (s_deserializer.deserialize(rawValue, sll) == Deserialization_SUCCESS)
+        {
+            switch (m_mode)
+            {
+            case Append:
+                if (!txSummary->append(m_info.m_name, sll, m_info.m_level, m_info.m_groups))
+                    UWARNLOG("script operation %s did not append '%s=%s'; name is malformed or is in use", m_tagname.str(), m_info.m_name.str(), rawValue);
```

Comment:
I'd prefer it if these warnings mentioned the type as well as the mode.  It's really valuable when you look at a line in the trace output or a warning from anywhere and can find the exact line in source that wrote it.

Reply:
Done.